packages/backend/src/open_payments/quote/service.ts

```diff
@@ -288,77 +295,58 @@ export async function startQuote(
   }
 }
 
-type QuoteHeaders = {
-  Accept: string
-  'Content-Type': string
-  'Rafiki-Signature'?: string
+function getFees(fee: Fee | undefined, principal: bigint): bigint {
+  const basisPoints = fee?.basisPointFee ?? 0
+  const fixedFee = fee?.fixedFee ?? 0n
+  const feePercentage = basisPoints / 10_000
+
+  // TODO: bigint/float multiplication
+  return BigInt(Math.floor(Number(principal) * feePercentage)) + fixedFee
 }
 
-export async function finalizeQuote(
+async function finalizeQuote(
   deps: ServiceDependencies,
   quote: Quote,
   receiver: Receiver,
   maxReceiveAmountValue?: bigint
 ): Promise<Quote> {
-  const requestHeaders: QuoteHeaders = {
-    Accept: 'application/json',
-    'Content-Type': 'application/json'
-  }
-
-  const body = {
-    ...quote.toJSON(),
-    paymentType: maxReceiveAmountValue
-      ? Pay.PaymentType.FixedSend
-      : Pay.PaymentType.FixedDelivery
-  }
-
-  if (deps.signatureSecret) {
-    requestHeaders['Rafiki-Signature'] = generateQuoteSignature(
-      body,
-      deps.signatureSecret,
-      deps.signatureVersion
-    )
-  }
-
-  const res = await axios.post(deps.quoteUrl, body, {
-    headers: requestHeaders,
-    validateStatus: (status) => status === 201
-  })
+  let debitAmountValue = quote.debitAmount.value
+  let receiveAmountValue = quote.receiveAmount.value
 
-  // TODO: validate res.data is quote
-  if (!res.data.sendAmount?.value || !res.data.receiveAmount?.value) {
-    throw QuoteError.InvalidAmount
-  }
-  let sendAmount: Amount
-  let receiveAmount: Amount
-  try {
-    sendAmount = parseAmount(res.data.sendAmount)
-    receiveAmount = parseAmount(res.data.receiveAmount)
-  } catch (_) {
-    throw QuoteError.InvalidAmount
-  }
-  if (maxReceiveAmountValue) {
+  if (!maxReceiveAmountValue) {
+    // FixedDelivery
+    const fees = getFees(quote.fee, debitAmountValue)
+    debitAmountValue = BigInt(debitAmountValue) + fees
     if (
-      sendAmount.value !== quote.sendAmount.value ||
-      receiveAmount.value > maxReceiveAmountValue ||
-      receiveAmount.value <= BigInt(0)
+      debitAmountValue < quote.debitAmount.value ||
+      debitAmountValue <= BigInt(0)
```

Comment:
How would that happen?

Reply:
I didn't realize it was just a refactor. I also looked more closely now and I think you are right. This can probably be removed since it checks the quote after it is returned from the ASE in the current flow.

Reply:
I could remove this one OK but I tried the same for fixed send and an existing test failed (which was explicitly looking for this case). Perhaps the inputs _shouldn't_ cause these errors but I think it's probably safer to leave them just in case.